test/features/auth/presenters/login_presenter_test.dart

```diff
@@ -1,28 +1,85 @@
+import 'package:flutter_demo/core/domain/model/user.dart';
+import 'package:flutter_demo/features/auth/domain/model/log_in_failure.dart';
+import 'package:flutter_demo/features/auth/domain/use_cases/log_in_use_case.dart';
 import 'package:flutter_demo/features/auth/login/login_initial_params.dart';
 import 'package:flutter_demo/features/auth/login/login_presentation_model.dart';
 import 'package:flutter_demo/features/auth/login/login_presenter.dart';
+import 'package:flutter_demo/localization/app_localizations_utils.dart';
 import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
 
+import '../../../test_utils/test_utils.dart';
 import '../mocks/auth_mock_definitions.dart';
+import '../mocks/auth_mocks.dart';
 
 void main() {
   late LoginPresentationModel model;
   late LoginPresenter presenter;
   late MockLoginNavigator navigator;
+  late LogInUseCase logInUseCase;
 
   test(
-    'sample test',
+    'when password and username arent provided then button is enabled',
     () {
-      expect(presenter, isNotNull); // TODO implement this
+      model.copyWith(userName: "", password: "");
+
+      expect(model.isLoginEnabled, false);
```

Comment:
you're not doing anything with the result of copying the model, the statement is completely redundant
```suggestion
      expect(model.copyWith(userName: '', password: '').isLoginEnabled, isFalse);
```